enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,830 @@
+* **Cluster Default Network** - This is the routed OVN network that pods attach to by default today as their primary network.
+The pods default route, service access, as well as kubelet probe are all served by the interface (typically eth0) on this network.
+* **User-Defined Network** - A network that may be primary or secondary, but is declared by the user.
+
+## Motivation
+
+As users migrate from OpenStack to Kubernetes, there is a need to provide network parity for those users. In OpenStack,
+each tenant (akin to Kubernetes namespace) by default has a layer 2 network, which is isolated from any other tenant.
+Connectivity to other networks must be specified explicitly as network configuration via a Neutron router. In Kubernetes
+the paradigm is opposite, by default all pods can reach other pods, and security is provided by implementing Network Policy.
+Network Policy can be cumbersome to configure and manage for a large cluster. It also can be limiting as it only matches
+TCP, UDP, and SCTP traffic. Furthermore, large amounts of network policy can cause performance issues in CNIs. With all 
+these factors considered, there is a clear need to address network security in a native fashion, by using networks per
+tenant to isolate traffic.
+
+### Goals
+
+* Provide a configurable way to indicate that a pod should be connected to a user-defined network of a specific type as a
+primary interface.
+* The primary network may be configured as a layer 3 or layer 2 type network.
+* Allow networks to have overlapping pod IP address space. This range may not overlap with the default cluster subnet
+used for allocating pod IPs on the cluster default network today.
+* The cluster default primary network defined today will remain in place as the default network pods attach to. The cluster
+default network will continue to serve as the primary network for pods in a namespace that has no user-defined network. Pods
+with user-defined networks will still attach to the cluster default network with limited access to Kubernetes system resources.
+Pods with user-defined networks will have at least two network interfaces, one connected to the cluster default network and one
+connected to the user-defined network.
+* Allow multiple namespaces per network.
+* Support cluster ingress/egress traffic for user-defined networks, including secondary networks.
+* Support for ingress/egress features where possible:
+    * EgressQoS
+    * EgressService
+    * EgressIP
+    * Load Balancer and NodePort Services, as well as services with External IPs.
+* In addition to ingress service support, there will be support for Kubernetes services in user-defined networks. The
+scope of reachability to that service as well as endpoints selected for that service will be confined to the network
+and corresponding namespace(s) where that service was created.
+* Support for pods to continue to have access to the cluster default primary network for DNS and KAPI service access.
```

Comment:
Ok yep, now I see it's outlined here that there's still k8s api access on the cluster default primary network